services/venice-controller/src/main/java/com/linkedin/venice/controller/server/ControllerRoutes.java

```diff
@@ -258,22 +258,11 @@ public Route getAdminOperationVersionFromControllers(Admin admin) {
       response.type(HttpConstants.JSON);
       try {
         String clusterName = request.queryParams(CLUSTER);
-        String currentUrl = getRequestURL(request);
-
         responseObject.setCluster(clusterName);
-        Map<String, Long> controllerUrlToVersionMap = admin.getAdminOperationVersionFromControllers(clusterName);
-        responseObject.setControllerUrlToVersionMap(controllerUrlToVersionMap);
-        responseObject.setRequestUrl(currentUrl);
-
-        if (controllerUrlToVersionMap.containsKey(currentUrl)) {
-          responseObject.setLocalAdminOperationProtocolVersion(controllerUrlToVersionMap.get(currentUrl));
-        } else {
-          // Should not happen
-          throw new VeniceException(
-              "The current controller URL: " + currentUrl + " is not in the urlToVersionMap in the response "
-                  + controllerUrlToVersionMap);
-        }
-
+        Map<String, Long> controllerNameToVersionMap = admin.getAdminOperationVersionFromControllers(clusterName);
+        responseObject.setControllerNameToVersionMap(controllerNameToVersionMap);
+        responseObject.setLocalControllerName(getControllerName(request));
```

Comment:
It seems odd to rely on an external request to figure out its own controller name.
Referring to the controllerName in VeniceHelixAdmin is more appropriate, but I guess it's always using the HTTP port for now? If that's the reason we are using external request as source of truth, let's add a TODO here as well as a tracking ticket to mention the long term plan - we will report the HTTPS port only to Helix, and revamp all tests to HTTPS only.

Reply:
Found the issue in this code. I fix the code to always reference the controllerName. 
When we turn off SSL, getControllerName from admin should give us the correct distinct name between controllers.